WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -0,0 +1,617 @@
+        return title
+    }
+    
+    func configureAuthorFilter() {
+        // Noop
+    }
+    
+    // MARK: - Sync Methods
+    
+    override internal func postTypeToSync() -> String {
+        return PostServiceTypePage
+    }
+    
+    override internal func lastSyncDate() -> NSDate? {
+        return blog?.lastPagesSync
+    }
+    
+    // MARK: - TableView Handler Delegate Methods
+    
+    func entityName() -> String {
+        return String(Page.self)
+    }
+    
+    
+    func predicateForFetchRequest() -> NSPredicate {
+        var predicates = [NSPredicate]()
+        
+        if let blog = blog {
+            let basePredicate = NSPredicate(format: "blog = %@ && original = nil", blog)
+            predicates.append(basePredicate)
+        }
+
+        let searchText = currentSearchTerm()
+        var filterPredicate = currentPostListFilter()?.predicateForFetchRequest
+        
+        // If we have recently trashed posts, create an OR predicate to find posts matching the filter,
+        // or posts that were recently deleted.
+        if let recentlyTrashedPostObjectIDs = recentlyTrashedPostObjectIDs
+            where searchText?.characters.count == 0 && recentlyTrashedPostObjectIDs.count > 0 {
+            
+            let trashedPredicate = NSPredicate(format: "SELF IN %@", recentlyTrashedPostObjectIDs)
+            
+            if let originalFilterPredicate = filterPredicate {
+                filterPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [originalFilterPredicate, trashedPredicate])
+            } else {
+                filterPredicate = trashedPredicate
+            }
+        }
+        
+        if let filterPredicate = filterPredicate {
+            predicates.append(filterPredicate)
+        }
+        
+        if let searchText = searchText where searchText.characters.count > 0 {
+            let searchPredicate = NSPredicate(format: "postTitle CONTAINS[cd] %@", searchText)
+            predicates.append(searchPredicate)
+        }
+        
+        let predicate = NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        return predicate
+    }
+    
+    // MARK: - Table View Handling
+
+    func sectionNameKeyPath() -> String {
+        return NSStringFromSelector(#selector(Page.sectionIdentifier))
+    }
+    
+    func tableView(tableView: UITableView, estimatedHeightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return CGFloat(self.dynamicType.pageCellEstimatedRowHeight)
+    }
+    
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        
+        if let page = tableViewHandler?.resultsController.objectAtIndexPath(indexPath) as? Page {
+            if cellIdentifierForPage(page) == self.dynamicType.restorePageCellIdentifier {
+                return CGFloat(self.dynamicType.pageCellEstimatedRowHeight)
+            }
+            
+            let width = CGRectGetWidth(tableView.bounds)
```

Comment:
`tableView.bounds.width`. CGRect is a struct, with few helper methods / shortcuts.